x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlAssignmentNotifier.java

```diff
@@ -136,48 +144,71 @@ private void auditMlTasks(
-                                    + currentAssignment.getExplanation()
-                                    + "]"
-                            );
+                            if (anomalyDetectionAuditor.includeNodeInfo()) {
+                                anomalyDetectionAuditor.warning(
+                                    jobId,
+                                    "No node found to start datafeed ["
+                                        + datafeedParams.getDatafeedId()
+                                        + "]. Reasons ["
+                                        + currentAssignment.getExplanation()
+                                        + "]"
+                                );
+                            } else {
+                                anomalyDetectionAuditor.warning(jobId, "Awaiting capacity to start datafeed.");
+                            }
                         } else if (wasTaskAssigned) {
-                            String nodeName = nodeName(previousNodes, previousAssignment.getExecutorNode());
-                            anomalyDetectionAuditor.info(
-                                jobId,
-                                "Datafeed [" + datafeedParams.getDatafeedId() + "] unassigned from node [" + nodeName + "]"
-                            );
+                            if (anomalyDetectionAuditor.includeNodeInfo()) {
+                                String nodeName = nodeName(previousNodes, previousAssignment.getExecutorNode());
+                                anomalyDetectionAuditor.info(
+                                    jobId,
+                                    "Datafeed [" + datafeedParams.getDatafeedId() + "] unassigned from node [" + nodeName + "]"
+                                );
+                            } else {
+                                anomalyDetectionAuditor.info(jobId, "Datafeed [" + datafeedParams.getDatafeedId() + "] relocating.");
+                            }
                         } else {
-                            logger.warn(
-                                "[{}] No node found to start datafeed [{}]. Reasons [{}]",
-                                jobId,
-                                datafeedParams.getDatafeedId(),
-                                currentAssignment.getExplanation()
-                            );
+                            if (anomalyDetectionAuditor.includeNodeInfo()) {
+                                logger.warn(
+                                    "[{}] No node found to start datafeed [{}]. Reasons [{}]",
+                                    jobId,
+                                    datafeedParams.getDatafeedId(),
+                                    currentAssignment.getExplanation()
+                                );
+                            } else {
+                                logger.warn("[{}] Failed to start datafeed [{}].", jobId, datafeedParams.getDatafeedId());
+                            }
```

Comment:
This one is just logging, not audit messages that will appear in the UI. I think it's OK to keep node info in our logging, because we're the primary people looking at those and it will be useful to us in diagnosing bugs. So please revert this small section to just the original log message.